docs/dev/design/sphinx-jquery.rst

```diff
@@ -0,0 +1,125 @@
+
+
+Calling ``app.setup_extension("sphinxcontrib.jquery")``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When a Sphinx theme or extension calls `setup_extension() <https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.setup_extension>`_, a call to ``sphinxcontrib.jquery.setup(app)`` will happen. Adding ``sphinxcontrib.jquery`` to a documentation project's ``conf.extensions`` will also call ``sphinxcontrib.jquery.setup(app)`` (at most once).
+
+In ``sphinxcontrib.jquery.setup(app)``, jQuery is added. The default behaviour is to detect the Sphinx version and include jQuery via `app.add_js_file <https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_js_file>`__ when Sphinx is from version 6 and up. jQuery is added at most once.
+
+
+Config value: ``jquery_force_enable``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When setting ``jquery_force_enable=True``, jQuery is added no matter the Sphinx version, but at most once. This is useful if you want to handle alternative conditions for adding jQuery.
+
+.. warning::
+
+  If you set ``jquery_force_enable=True``, you most likely should also add ``Sphinx>=6`` to your theme's/extension's dependencies since versions before this already bundles jQuery!
+
+
+Release
+-------
+
+All releases follow semantic versioning: ``<major>.<minor>.<patch>``
+
+* ``major``: Changes that break existing themes and extensions - we'll aim to only ever have ``1.x.y`` releases!
+* ``minor``: Changes that contain jQuery updates or substantial feature additions.
+* ``patch``: Bug fixes and added compatibility.
+
+For distributed packages with a longer lifecycle, we propose a loose pinning of the patch and minor versions to allow for future updates, which are likely to happen when jQuery receives security updates.
+Example: ``sphinxcontrib-jquery>=1.0.0,<2.0.0`` will include all compatible future releases, including jQuery upgrades.
+
+
+jQuery version and inclusion
+----------------------------
+
+Sphinx has kept relatively up to date with jQuery, and this package intends to follow.
+The most recently bundled jQuery version was v3.5.1 and only two releases have happened since: 3.6.0 and 3.6.1.
+The 3.6.0 release had a very small backwards incompatibility which illustrates how harmless these upgrades are for the general purpose Sphinx package.
+
+Therefore, we propose to start 1.0.0 at 3.5.1 (the currently shipped version) and subsequently release 3.6.1 as the first update of jQuery as 1.1.0.
+
+The bundled jQuery version will be NPM pre-minified and distributed together with the PyPI package.
+
+The minified jQuery JS file is ultimately included by calling `app.add_js_file <https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_js_file>`__, which is passed the following arguments:
+
+.. code:: python
+
+  app.add_js_file(
+      get_jquery_url_path(),
+      loading_method="defer",
+      priority=200,
```

Comment:
Where this 200 as priority comes from? Is it 200 for some specific reason?

Reply:
See https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_js_file

Reply:
Also, I think that 200 is the correct value.

Reply:
From that link, I'd say that we should use `500: default priority for extensions` since `200: default priority for built-in JavaScript files` and this is not a built-in javascript (it _was_, tho).